### ISSUE TITLE
Email + password login not allowed due to email signup being disabled

### ISSUE THREAD

**Issue by TomasHubelbauer** [supabase/supabase#4746]
# Bug report

## Describe the bug

It is possible to sign up a new user using the `supabase.auth.signUp` method. Once they are signed up, disabling the below pictured option prevents other usings from signing up to the service:
<img width="806" alt="image" src="https://user-images.githubusercontent.com/6831144/147787314-a20d18b9-7657-43e8-8b9f-e47979ce6936.png">

This is great as it allows service administrators to keep the service open to a limited set of users. They may even choose to never open any option of external signup and only manually invite users through the dashboard.

However, with this option disabled, logging in of existing users ceases to work. Both in case of magic link and a password based token grant flow. The endpoint being called by supabase-js fails with a 400:
<img width="183" alt="image" src="https://user-images.githubusercontent.com/6831144/147787404-749ed5b2-a371-497b-9aad-fd26b0ac5da6.png">

This response is the same for both magic link (endpoint is `auth/v1/magiclink`) and password based token auth (endpoint is `auth/v1/token?grant_type=password`).

## To Reproduce

1. Enable email signup in the dashboard
2. Call `supabase.auth.signUp({ email, password })` and click the verification link in the email
3. Disable email signup in the dashboard
4. Call `supabase.auth.signIn({ email, password })`
5. Observe a 400 non-OK response and `supabase.auth.user()` returning `null`
6. Enable email signup in the dashboard
7. Call `supabase.auth.signIn({ email, password })`
8. Observe a 200 OK response and `supabase.auth.user()` returning an authorized user object

## Expected behavior

User sign in should work regardless of the ability of new users to sign up. Site operators may choose to cease accepting registrations while still providing service to existing accounts.

## System information

- Version of supabase-js: 1.25.2

## Additional context

Related discussion: https://github.com/supabase/supabase/discussions/4296

This is the problematic line: https://github.com/supabase/gotrue/blob/master/api/token.go#L157

I believe the fix might be to just yeet the whole condition, but I am not able to determine whether that might break something else? if anyone can confirm this condition can be safely removed without any adverse effects, I will be more than happy to PR.

I opened the issue here as from the PoV of the Supabase user this is the SDK misbehaving, but an argument could be made this is an issue better suited for supabase/gotrue, so please move it as appropriate.

**Comment by kiwicopple** [comment]
Yeah I would not have expected this to block logins too. I'll transfer this over to GoTrue - thanks for reporting @TomasHubelbauer
